modules/calc/src/main/java/com/opengamma/strata/calc/marketdata/DefaultMarketDataFactory.java

```diff
@@ -166,21 +165,21 @@ public MarketEnvironment buildMarketData(
 
       // Build any time series that are required but not available
       leafRequirements.getTimeSeries().stream()
-          .filter(not(marketData::containsTimeSeries))
-          .filter(not(suppliedData::containsTimeSeries))
+          .filter(id -> marketData.getTimeSeries(id).isEmpty())
+          .filter(id -> suppliedData.getTimeSeries(id).isEmpty())
```

Comment:
The change in meaning seems significant to this logic. No way to prevent query for time-series just by putting empty one in input.